design/design-doc-integration-tests.md

```diff
@@ -84,21 +83,43 @@ The framework should be made up of two core components:
 
 #### Jobs
 
-A `Job` contains the following components. `*Config` will likely evolve over
-time, but will start with just defining the `SyncPeriod` for the controller
-`manager`.
+A `Job` contains the following components:
 
 ```go
-// A Job is a set of tests that run in a custom configured local Kubernetes environment.
+// A Job is a set of tests that are executed sequentially in the same cluster environment
 type Job struct {
-    name             string
-    description      string
-    cfg              *Config
-    runner           *testing.T
-    setupWithManager setupWithManagerFunc
-    addToScheme      addToSchemeFunc
-    jobClean         jobCleanFunc
-    tests            []Test
+    Name        string
+    Description string
+    Tests       []Test
+    cfg         *JobConfig
+    runner      *testing.T
+}
+```
+
+A `JobConfig` is made up of optional values that are applied via `JobOption`
+functions:
+
+```go
+// JobConfig is a set of configuration values for a Job
+type JobConfig struct {
+    CRDDirectoryPaths []string
+    Cluster           *rest.Config
+    Builder           OperationFn
+    Cleaner           OperationFn
+    SetupWithManager  SetupWithManagerFunc
+    AddToScheme       AddToSchemeFunc
+    SyncPeriod        *time.Duration
+}
+```
+
+An example of a `JobOption` function:
+
+```go
+// WithCRDDirectoryPaths sets custom CRD locations for a Job
+func WithCRDDirectoryPaths(crds []string) JobOption {
+    return func(j *Job) {
+        j.cfg.CRDDirectoryPaths = crds
```

Comment:
I think that would make sense here. I chose to operate on `*Job` because it is more flexible for future implementation. However, if we did go with operating on `*JobConfig` for now any future change to operating on `*Job` would not actually require code changes from an end-user perspective, so I would be inclined to go with your recommendation here. Thanks!